pkgs/development/libraries/wolfssl/default.nix

```diff
@@ -6,73 +6,102 @@
 , util-linux
 , openssl
 }:
-
-stdenv.mkDerivation rec {
-  pname = "wolfssl";
-  version = "5.6.3";
-
-  src = fetchFromGitHub {
+let
+  defaultVersion = "5.6.3";
+  defaultSrc = fetchFromGitHub {
     owner = "wolfSSL";
     repo = "wolfssl";
-    rev = "refs/tags/v${version}-stable";
+    rev = "refs/tags/v${defaultVersion}-stable";
     hash = "sha256-UN4zs+Rxh/bsLD1BQA+f1YN/UOJ6OB2HduhoetEp10Y=";
   };
 
-  postPatch = ''
-    patchShebangs ./scripts
-    # ocsp tests require network access
-    sed -i -e '/ocsp\.test/d' -e '/ocsp-stapling\.test/d' scripts/include.am
-    # ensure test detects musl-based systems too
-    substituteInPlace scripts/ocsp-stapling2.test \
-      --replace '"linux-gnu"' '"linux-"'
-  '';
+  buildWolfSSL = lib.makeOverridable
```

Comment:
Do you _actually_ want to do it this way? Adding a third level of overridability?

Which is the bit of this that's not doable using a combination of the existing `callPackage`-level `.override` and `.overrideAttrs`?

Reply:
I'm indecisive, really.

My thought was to make it as ergonomic as possible to override the derivations, as the variants will grow a lot when more support is added to specific builds and one might want to override the feature flags quickly to add specific optimizations and such.

But I think the `overrideAttrs` could suffice now that `buildWolfSSL` exists in general and one isn't forced to respecify all flags.
Adding flags will be a bit more verbose, but if you prefer to not add this extra layer, I'll remove that bit.

Reply:
Are you aware of the `finalAttrs`/`finalPackage` mechanism? It can be used for some quite powerful patterns that don't involve an intermediary `build*` function (because `.override` and `.overrideAttrs` are made to do the right thing for most cases)